tests/prompts.test.ts

```diff
@@ -0,0 +1,398 @@
+        })
+      );
+      expect(prompt).toMatchObject({
+        fandoms: ["陈情令 | The Untamed (TV)"],
+        tags: {
+          warnings: ["Creator Chose Not To Use Archive Warnings"],
+          characters: [],
+          relationships: ["Lan Huan | Lan Xichen/Meng Yao | Jin Guangyao"],
+          additional: ["Crack Treated Seriously"]
+        }
+      });
+    });    
+
+
+
+
+
+    test("Fetches claims:  0 claims", async () => {
+      const prompt = await getPrompt(
+        await getPromptDetailsFromUrl({url: "https://archiveofourown.org/collections/test_prompt_meme_2024/prompts/3573835"})
+      );
+      expect(prompt).toMatchObject({
+        claims: {count: 0},
+      });
+    });      
+
+    test("Fetches claims: Anon Claims + No Known Claimaints", async () => {
+      const prompt = await getPrompt(
+        await getPromptDetailsFromUrl({url: "https://archiveofourown.org/collections/mo_dao_zu_shi_kink_meme_2020/prompts/1909048"})
+      );
+      expect(prompt).toMatchObject({
+        claims: {count: 3, isAnonCollection: true},
+      });
+    });  
+
+    test("Fetches claims: Known Claimaints+No Anon Claimaints", async () => {
+      const prompt = await getPrompt(
+        await getPromptDetailsFromUrl({url: "https://archiveofourown.org/collections/test_prompt_meme_2024/prompts/3583348"})
+      );
+      expect(prompt).toMatchObject({
+        claims: {count: 2, isAnonCollection: false, claimantUsernames:["RabbitPie","enigmalea"]},
+      });
+    });  
+ 
+
+    //WIP PROMPTS
+    test("Fetches Prompt Categores (multiple)", async () => {
+      const prompt = await getPrompt(
+        await getPromptDetailsFromUrl({url: "https://www.archiveofourown.org/collections/test_prompt_meme_2024/prompts/3573835"})
+      );
+      expect(prompt).toMatchObject({
+        categories: ["M/M", "Other"],
+      });
+    });
+
+      test("Fetches Prompt Categores (single)", async () => {
+        const prompt = await getPrompt(
+          await getPromptDetailsFromUrl({url: "https://www.archiveofourown.org/collections/mo_dao_zu_shi_kink_meme_2020/prompts/1927806"})
+        );
+        expect(prompt).toMatchObject({
+          categories: ["Gen"],
+        });  
+      });    
+
+    test("Fetches Prompt Categores (0 as null) with multiple fills", async () => {
+      const prompt = await getPrompt(
+        await getPromptDetailsFromUrl({url: "http://www.archiveofourown.org/collections/mo_dao_zu_shi_kink_meme_2020/prompts/1909048"})
+      );
+      expect(prompt).toMatchObject({
+        categories: null,
+      });
+    });      
+
+    test("Fetches collectionName & PromptID", async () => {
+      const prompt = await getPrompt(
+        await getPromptDetailsFromUrl({url: "http://www.archiveofourown.org/collections/mo_dao_zu_shi_kink_meme_2020/prompts/1909048"})
+      );
+      expect(prompt).toMatchObject({
+        collectionName: "mo_dao_zu_shi_kink_meme_2020",
+        id: "1909048"
+      });
+    }); 
+
+  });
+
+
```

Comment:
Suggestion: i think this file is getting long so it might be worth considering splitting it in multiple files